chore(verification): update README and reorganize generate_report.py (#1978)
meta-llama/llama-stack: tests/verifications/README.md

## Features

The verification suite currently tests the following in both streaming and non-streaming modes:

- Basic chat completions
- Image input capabilities
- Structured JSON output formatting
- Tool calling functionality

## Report

The lastest report can be found at [REPORT.md](REPORT.md).

To update the report, install the dependencies
```bash

> pip install -e .

Reviewer: remove this if we use `uv`

```
ensure you have the API keys set,
```bash
export OPENAI_API_KEY=<your_openai_api_key>
export FIREWORKS_API_KEY=<your_fireworks_api_key>
export TOGETHER_API_KEY=<your_together_api_key>
```
then run
```bash

> python tests/verifications/generate_report.py --run-tests

Reviewer: ```suggestion
uv run -with-editable . --with-editable ".[dev]" --with-editable ".[unit]" python tests/verifications/generate_report.py --run-tests
```

Which deps do you need?

> The verification suite currently tests:

Reviewer 2: ```suggestion
The verification suite currently tests the following in both streaming and non-streaming modes:
```

- Basic chat completions
- Image input capabilities
- Structured JSON output formatting
- Tool calling functionality

## Report

> The lastest report can be found at [REPORT.md](https://github.com/meta-llama/llama-stack/blob/main/tests/verifications/REPORT.md).

Reviewer 2: ```suggestion
The latest report can be found at [REPORT.md](REPORT.md).
```

To update the report, run
```bash

> python tests/verifications/generate_report.py --run-tests

Reviewer 2: will be good to include steps to set things up: `pip install -r requirements` or maybe just `uv run --with openai tests/verifications/generate_report.py`?

Also, list out the API keys needed to be set - is OPENAI_API_KEY enough for any provider? And then, it will get mapped to the appropriate env var needed by the provider

Reviewer: I also think we should mention `uv` since all our guides recommend it.
